docs/proposals/extracted_credentials_saved_as_secrets.md

```diff
@@ -0,0 +1,59 @@
+## Extracted Credentials Saved As Secrets
+
+Extracted Credentials are currently saved in our etcd for the service broker. This is not desirable for many reasons, but the two biggest are kubernetes already has a built-in way to manage this data, secrets, and when moving to CRDs we don't want to create a resource for extracted credentials.
+
+### Problem Description
+The problem is that we should not manage data that is of a sensitive nature if we do not have to. This proposal is limited in scope and only interested in how we save the extracted credentials. It is worth noting that we should eventually be better about how we transmit this data to APBs. 
+
+In the secret, we will save the data in the following format.
+```yaml
+data:
+  DB_PASSWORD: <GOB_ENCODED_VALUE>
+  DB_USERNAME: <GOB_ENCODED_VALUE>
+  ....
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <Service/Binding id>
+  namespace: <Namespace for the broker>
+  labels:
+    <labels provided>
+```
+
+[Gob encoding](https://godoc.org/encoding/gob) will allow us to save arbitrary data in the secret for a key. The secrets keys will look rational to a user who looks at the created secret. This user would need permissions to see the secret, but if someone is looking at the secret making it obvious what data is in there will be helpful. 
+
+The functions for saving and retrieving will be in the `clients` package. This means the callers will be required to use the underlying extracted credentials type `map[string]interface{}` because we do not want a circular dependency between `apb` package and `clients` package. 
+
+We will interact with the secrets from the namespace defined in the configuration by the `openshift.namespace` value. 
```

Comment:
It occurs to me reading this that we really should remove references to `openshift` from our configuration if we're looking to be cluster agnostic. Not at all in the scope of this proposal, just thought I'd call it out. We probably need an issue to track it.